src/YesSql.Provider.MySql/MySqlDbProviderOptionsExtensions.cs

```diff
@@ -31,8 +42,18 @@ public static class MySqlDbProviderOptionsExtensions
             }
 
             configuration.ConnectionFactory = new DbConnectionFactory<MySqlConnection>(connectionString);
-            configuration.DocumentStorageFactory = new SqlDocumentStorageFactory();
             configuration.IsolationLevel = isolationLevel;
+
+            if (cached)
+            {
+                configuration.DocumentStorageFactory = 
+                    new CacheDocumentStorageFactory(new SqlDocumentStorageFactory());
+            }
+            else
+            {
+                configuration.DocumentStorageFactory = 
+                    new CacheDocumentStorageFactory(new SqlDocumentStorageFactory());
```

Comment:
Wait a minute I think I made a big mistake here before I submit the PR, I will update it now